src/wp-includes/block-editor.php

```diff
@@ -243,6 +243,41 @@ function get_block_editor_settings( $editor_name, $custom_settings = array() ) {
 		$custom_settings
 	);
 
+	$editor_settings['__experimentalFeatures'] = WP_Theme_JSON_Resolver::get_merged_data( $editor_settings )->get_settings();
+
+	// These settings may need to be updated based on data coming from theme.json sources.
+	if ( isset( $editor_settings['__experimentalFeatures']['color']['palette'] ) ) {
+		$editor_settings['colors'] = $editor_settings['__experimentalFeatures']['color']['palette'];
```

Comment:
For the top-level colors the editor is going to use $editor_settings['colors'] or $editor_settings['__experimentalFeatures']['color']['palette']? I guess the editor uses $editor_settings['__experimentalFeatures']['color']['palette'] right?
In that case if a plugin is filtering the colors in $editor_settings['colors'] that plugin will stop working?